homeassistant/components/eq3btsmart/climate.py

```diff
@@ -0,0 +1,320 @@
+"""Platform for eQ-3 climate entities."""
+
+import logging
+from typing import Any
+
+from eq3btsmart import Thermostat
+from eq3btsmart.const import EQ3BT_MAX_TEMP, EQ3BT_OFF_TEMP, Eq3Preset, OperationMode
+from eq3btsmart.exceptions import Eq3Exception
+
+from homeassistant.components.climate import (
+    ATTR_HVAC_MODE,
+    PRESET_NONE,
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACAction,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, PRECISION_HALVES, UnitOfTemperature
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import ServiceValidationError
+from homeassistant.helpers.device_registry import (
+    CONNECTION_BLUETOOTH,
+    DeviceInfo,
+    async_get,
+    format_mac,
+)
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import (
+    DEVICE_MODEL,
+    DOMAIN,
+    EQ_TO_HA_HVAC,
+    HA_TO_EQ_HVAC,
+    MANUFACTURER,
+    SIGNAL_THERMOSTAT_CONNECTED,
+    SIGNAL_THERMOSTAT_DISCONNECTED,
+    CurrentTemperatureSelector,
+    Preset,
+    TargetTemperatureSelector,
+)
+from .entity import Entity
+from .models import Eq3Config, Eq3ConfigEntryData
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Handle config entry setup."""
+
+    eq3_config_entry: Eq3ConfigEntryData = hass.data[DOMAIN][config_entry.entry_id]
+
+    async_add_entities(
+        [Eq3Climate(eq3_config_entry.eq3_config, eq3_config_entry.thermostat)],
+    )
+
+
+class Eq3Climate(Entity, ClimateEntity):
+    """Climate entity to represent a eQ-3 thermostat."""
+
+    _attr_has_entity_name: bool = True
+    _attr_name: str | None = None
```

Comment:
```suggestion
    _attr_name = None
```

typing is already set by the base class. 

Please remove redundant type hints

Reply:
My IDE does not understand this, but I'll remove them.

![grafik](https://github.com/home-assistant/core/assets/47421406/156964c7-b358-4541-9837-1c413e998dc5)